lib/modules/manager/cargo/extract.ts

```diff
@@ -286,9 +289,41 @@ export async function extractPackageFile(
     'Cargo.lock',
   );
   const res: PackageFileContent = { deps };
-  // istanbul ignore if
   if (lockFileName) {
+    logger.debug(
+      `Found lock file ${lockFileName} for packageFile: ${packageFile}`
+    );
+
+    const versionsByPackage = await extractLockFileVersions(lockFileName);
+    if (!versionsByPackage) {
+      logger.debug(
+        `Could not extract lock file versions from ${lockFileName}.`
+      );
+      return res;
+    }
+
     res.lockFiles = [lockFileName];
+
+    for (const dep of deps) {
+      const packageName = dep.packageName ?? dep.depName!;
+      const versions = coerceArray(versionsByPackage.get(packageName));
+      try {
+        const lockedVersion = versioning.getSatisfyingVersion(
+          versions,
+          dep.currentValue!
+        );
+        if (lockedVersion) {
+          dep.lockedVersion = lockedVersion;
+        }
```

Comment:
Should we log a debug message if there's an "else"? Wouldn't it mean that there's no locked version for a dependency?